tools/hangar/__snapshots__/invalid.ts.snap

```diff
@@ -212,12 +212,14 @@ Duration <DURATION>"
 `;
 
 exports[`capture_redefinition.w 1`] = `
-"error: Cannot capture symbol \\"y\\" because it is shadowed by another symbol with the same name
-   --> ../../../examples/tests/invalid/capture_redefinition.w:14:9
-   |
-14 |     log(y);
-   |         ^ Cannot capture symbol \\"y\\" because it is shadowed by another symbol with the same name
+"ERROR: Cannot access 'x' before initialization
```

Comment:
Figure out why we don't get these errors anymore and if this is a good thing?

Reply:
This seems like an unwanted regression

Reply:
Fixed by adding a dedicated test for var shadowing in the lifting phase by cleaning up the previous code and moving it to the right place. The new test is more in-line with how JS works so this actually produces the desired result in all three test cases in _capture_redefinition.w_. Added a big comment explaining the logic.

Reply:
Actually this still isn't perfect, need to better distinguish between _capture_redefinition.w_ and _use_before_defined.w_ tests..

Reply:
fixed by note checking for capturing a shadowed var for unresolved expressions (unknown symbols). This avoids cascading errors.